provisioning/ubuntu-provision.sh

```diff
@@ -64,6 +66,72 @@ function clean_apt() {
   apt-get upgrade -y
 }
 
+## Ensure that there is a user named "jenkins" created and configured
+function setuser() {
+  groupadd --gid="${groupid}" "${groupname}"
+
+  # jenkins should not be able to run sudo commands
+  useradd --create-home \
+    --home-dir "${userhome}" \
+    --uid "${userid}" \
+    --gid "${groupid}" \
+    --shell /bin/bash \
+    "${username}"
+
+  # Add authorized keys to jenkins user
+  mkdir -p "${userhome}/.ssh"
+  curl --fail --silent --location --show-error "${OPENSSH_AUTHORIZED_KEYS_URL}" --output "${userhome}/.ssh/authorized_keys"
+  chmod 0700 "${userhome}/.ssh"
+  chmod 0600 "${userhome}/.ssh/authorized_keys"
+  chown -R jenkins:jenkins "${userhome}/.ssh"
+}
+
+## Install asdf on the default user's home
+function install_asdf() {
+  local archive profile_script
+  archive=/tmp/asdf.tgz
+  profile_script="${userhome}/.bashrc"
+
+  if test -f "${userhome}/.asdf/asdf.sh"
+  then
+    return 0
+  fi
+  test -d "${userhome}"
+
+  curl --fail --silent --show-error --location "https://github.com/asdf-vm/asdf/archive/refs/tags/v${ASDF_VERSION}.tar.gz" --output "${archive}"
+  mkdir -p "${asdf_install_dir}"
+  tar --extract --verbose --gunzip --file="${archive}" --directory="${asdf_install_dir}" --strip-components=1 #strip the 1st-level directory of the archive as it has a changing name (asdf-<version>)
+
+  touch "${profile_script}"
+  echo ". ${asdf_install_dir}/asdf.sh" >> "${profile_script}"
+  chown -R "${username}:${groupname}" "${userhome}"
+  rm -f "${archive}"
+}
+
+## Install the ASDF Plugin passed as argument ($1 is the name and $2 the URL)
+function install_asdf_plugin() {
+  local plugin_name="${1}"
+  local plugin_url="${2}"
+
+  # Git is required to install asdf plugins
+  command -v git >/dev/null 2>&1
+
+  # sleep 10000
```

Comment:
?

Reply:
ah ah you caught me red-handed: forgot to remove my debug stances. Thanks Tim.